rocky/rocky/views/tasks.py

```diff
@@ -73,6 +80,31 @@ def get_queryset(self):
             messages.add_message(self.request, messages.ERROR, error_message)
             return []
 
+    def post(self, request, *args, **kwargs):
+        if "action" in self.request.POST:
+            self.handle_page_action(request.POST["action"])
+        return self.get(request, *args, **kwargs)
+
+    def handle_page_action(self, action: str):
+        if action == PageActions.RESTART_TASK.value:
+            task_id = self.request.POST.get("task_id")
+            task = client.get_task_details(task_id)
+
+            new_id = uuid.uuid4()
+
+            p_item = task.p_item
+            p_item.id = new_id
+            p_item.data.id = new_id
+
+            client.push_task(f"{task.type}-{self.organization.code}", task.p_item)
+
+            success_message = (
+                "Your task is running successfully in the background. \n "
+                "Results will be added to the object list when they are in. "
+                "It may take some time, a refresh of the page may be needed to show the results."
```

Comment:
The new task is scheduled, but doesn't have to be running the background yet.

Reply:
Fixed